src/neuroconv/datainterfaces/ecephys/openephys/openephyslegacydatainterface.py

```diff
@@ -0,0 +1,103 @@
+"""Authors: Szonja Weigl, Cody Baker."""
+from typing import Optional
+from warnings import warn
+
+from neo.rawio import OpenEphysRawIO
+from spikeinterface.extractors import OpenEphysLegacyRecordingExtractor
+
+from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
+from ....utils import FolderPathType, get_schema_from_method_signature
+
+
+class OpenEphysLegacyRecordingInterface(BaseRecordingExtractorInterface):
+    """Primary data interface for converting legacy Open Ephys data (.continuous files).
+    Uses :py:class:`~spikeinterface.extractors.OpenEphysLegacyRecordingExtractor`."""
+
+    ExtractorName = "OpenEphysLegacyRecordingExtractor"
+
+    @classmethod
+    def get_source_schema(cls):
+        """Compile input schema for the RecordingExtractor."""
+        source_schema = get_schema_from_method_signature(
+            class_method=cls.__init__,
+            exclude=["stream_id", "stream_name", "block_index", "all_annotations", "stub_test"],
+        )
+        source_schema["properties"]["folder_path"][
+            "description"
+        ] = "Path to directory containing OpenEphys legacy files."
+        return source_schema
+
+    def __init__(
+        self,
+        folder_path: FolderPathType,
+        stream_id: Optional[str] = None,
+        stream_name: Optional[str] = None,
+        block_index: Optional[int] = 0,
+        all_annotations: Optional[bool] = False,
+        stub_test: bool = False,
+        verbose: bool = True,
+    ):
+        """
+        Initialize reading of OpenEphys legacy recording (.continuous files).
+        See :py:class:`~spikeinterface.extractors.OpenEphysLegacyRecordingExtractor` for options.
+
+        Parameters
+        ----------
+        folder_path : FolderPathType
+            Path to OpenEphys directory.
+        stream_id : str, default: None
+            The identifier of the recording stream.
+            When the recording stream is not specified the channel stream ("CH") is chosen if available.
+            When channel stream is not available the stream_id or stream_name must be specified.
+        stream_name : str, default: None
+            The name of the recording stream.
+        block_index : int, default: 0
+        all_annotations : bool, default: False
+        stub_test : bool, default: False
+        verbose : bool, default: True
+        """
+        self.RX = OpenEphysLegacyRecordingExtractor
+
+        self.folder_path = folder_path
```

Comment:
This will be recorded automatically in the `self.source_data` dictionary so no need to set it explicitly

Reply:
Also, the only place it's used is in the private method that I'd like to see as a classmethod (and would therefore take `folder_path` as an input instead of as an attribute) so I think you can remove this